File: checkio_js/electronic_station/javascript/isAllUpper.js
import assert from "assert";

function isAllUpper(text) {
    return text.match(/[a-zA-Z]/) ? text === text.toUpperCase() : false
}

console.log('Example:');
console.log(isAllUpper('ALL UPPER'));

// These "asserts" are used for self-checking
assert.equal(isAllUpper('ALL UPPER'), true);
assert.equal(isAllUpper('all lower'), false);
assert.equal(isAllUpper('mixed UPPER and lower'), false);
assert.equal(isAllUpper(''), false);
assert.equal(isAllUpper("     "), false);

console.log("Coding complete? Click 'Check' to earn cool rewards!");

File: checkio_js/electronic_station/python/isAllUpper.py
import re

def isAllUpper(text):
    return text == text.upper() if re.findall(r'[a-zA-Z]', text) != [] else False

if __name__ == '__main__':
    print('Example:');
    print(isAllUpper('ALL UPPER'));

# These "asserts" are used for self-checking
assert isAllUpper('ALL UPPER') == True
assert isAllUpper('all lower') == False
assert isAllUpper('mixed UPPER and lower') == False
assert isAllUpper('') == False
assert isAllUpper("     ") == False

print("Coding complete? Click 'Check' to earn cool rewards!")
